### PR TITLE
Updated documentations + File log bug fix

### PR DESCRIPTION
**Pull request by cyrus2281**
- Added clarifications
    - Added instruction for creating a local solace broker
    - Update Makefile and Dockerfile with minor details
    - Fixed bug with writing stream file

### CODE REVIEW

File: src/solace_ai_connector/common/log.py [hunk 1]
```diff
@@ -18,9 +18,13 @@ def setup_log(logFilePath, stdOutLogLevel, fileLogLevel):
...
     stream_handler.setFormatter(stream_formatter)
 
+    # Create an empty file at logFilePath (this will overwrite any existing content)
+    with open(logFilePath, "w") as file:
+        file.write("")        
+
     # file_handler = logging.handlers.TimedRotatingFileHandler(
     #    filename=logFilePath, when='midnight', backupCount=30, mode='w')
-    file_handler = logging.FileHandler(filename=logFilePath, mode="w")
+    file_handler = logging.FileHandler(filename=logFilePath, mode="a")
```
Comment (cyrus2281) [inline src/solace_ai_connector/common/log.py:27]: Choose the write mode as `w` (`write`) would close the stream for any follow up log message. Changing to `a` (`append`) keeps the stream open for log stream (Added a manual write to file to clear content)

File: Dockerfile [hunk 2]
```diff
@@ -6,14 +6,16 @@ RUN apt-get update && \
...
 COPY requirements.txt /app
 
 RUN python3.10 -m pip install -r requirements.txt
 ENV PYTHONUNBUFFERED=1
 
-LABEL org.opencontainers.image.source https://github.com/SolaceLabs/solace-ai-connector
+#Install main program
+COPY /src /app/src
+
+LABEL org.opencontainers.image.source=https://github.com/SolaceLabs/solace-ai-connector
```
Comment (efunneko) [inline Dockerfile:18]: This is fine, but I am curious why the change?

**Review by efunneko — APPROVED** [review]
Looks good. Thanks!

File: Dockerfile (hunk 2 above)
Comment (cyrus2281) [reply Dockerfile:18]: So the spaced key/value is the legacy format, in the newer docker files all the ARG, LABEL, and ENV should be `key=value`, More: https://docs.docker.com/reference/build-checks/legacy-key-value-format/

Comment (cyrus2281) [reply Dockerfile:18]: If asking about why moving the COPY source code after the dependency installation, it's to best use the docker caching system, (Useful when using Docker for local development, no effect on CI/CD pipeline). Docker caches each stage (each line from the dockerfile) until the file being updated has changed, copying the source code after dep install allows not having to wait for dep installatIon after only changing the source code.
More: https://docs.docker.com/build/cache/#order-your-layers
